app/javascript/controllers/heart_controller.js

```diff
@@ -21,12 +22,29 @@ export default class extends Controller {
       this.animation.clickUnfave()
       this.favorited = false
       this.unfaved()
+      this.unfavore()
     } else {
       this.animation.clickFave()
       this.favorited = true
       this.faved()
+      this.favore()
     }
   }
+  unfavore() {
+    Rails.ajax({
+      url: `/${this.data.get('current-user')}/favorites/delete`,
```

Comment:
This opens up an interesting Q. We have `window.username` (which should probably be renamed to `.login` and eventually we should probably move that stuff into a "real" js object) — but is there a good reason for the route to include the user at all? See follows: https://github.com/sudara/alonetone/blob/master/config/routes.rb#L45-L46

Wondering if we can just do `/favorites/234` since when we are logged in, the user is always `current_user`

Just a question of what we want to be consistent with + what's simplest...

Reply:
> Wondering if we can just do /favorites/234 since when we are logged in, the user is always current_user

Few reasons:
- we need a user to assign user's favorites. So it was either by ensuring we get a `current_user` correctly in the controller or by using named route
- I read it more as user's favorites `sudara/favorites/123` rather than `/favorites/123`

But I see your point though. Looked over the code I removed, and it also just says `current_user.`... Let's try the simpler route.

Would you prefer to keep `favorites#create/delete` or should I make custom names as `put :favor` and `put :favor_not` (ha LOL ... `unfavor` just doesn't sound good) ... remove_favorite and add_favorite. Names are hard.

Reply:
I think I prefer favorites_controller#create/delete to favorite/unfavorite or add_favorite/remove_favorite.